models/user/error.go

```diff
@@ -107,3 +107,47 @@ func IsErrUserIsNotLocal(err error) bool {
 	_, ok := err.(ErrUserIsNotLocal)
 	return ok
 }
+
+// ErrBadgeAlreadyExist represents a "badge already exists" error.
```

Comment:
Putting the errors close to related structs is better.

Reply:
ok. updated.